### ISSUE TITLE
[FE] Direct Deposit: (Bug) Warning for unsaved bank info changes not always shown

### ISSUE THREAD

**Issue by erikphansen** [department-of-veterans-affairs/va.gov-team#25792]
When both bank info sections are in edit mode, there are situations where you can have unsaved changes in one form and you are not warned about losing that data if you attempt to change pages.

It seems like if you start making edits in both forms at the same time, then exit out of one edit form, the app is no longer aware of the fact that you have unsaved changes in the other edit form.

1. sign in with user 378
2. go to https://staging.va.gov/profile/direct-deposit
3. enter edit mode for the DD4CNP bank info
4. enter edit mode for the DD4EDU bank info
5. enter a routing number in DD4CNP
6. enter a routing number in DD4EDU
7. Cancel out of DD4CNP
8. navigate to another page in Profile
9. **You should be warned about losing unsaved changes to DD4EDU but you are not**

https://images.zenhubusercontent.com/133843125/56bddf05-649d-4a43-843f-d1c4f0199e91/no_warning.mp4

**Comment by ajakabcin** [comment]
@erikphansen and @tmitchellgcio let's revisit this ticket once #27097 is complete.

**Comment by tmitchellgcio** [comment]
Update on this ticket. Looks like I was doing this all wrong. The prompt is what we seem to care about showing up not the modal. I thought the issue was for both being open at the same time which still feels weird to me, but it's not. I think I misread this ticket. I'm on the right path now so I'm working on it.

**Comment by tmitchellgcio** [comment]
@ajakabcin should be ready for validation

**Comment by ajakabcin** [comment]
thanks @tmitchellgcio - I'm seeing the warning message now about losing unsaved edits, but I'm not sure the formatting of the alert is correct. Tagging @andaleliz for input. 

This is the alert you see when clicking cancel without saving after making edits/adding information:
<img width="767" alt="Screen Shot 2021-11-09 at 1 06 47 PM" src="https://user-images.githubusercontent.com/45603961/140980110-a17fd795-5cb2-427b-b9c2-c274dd12d500.png">
 
And this is the alert you see when attempting to navigate to another page in profile without saving: 
<img width="810" alt="Screen Shot 2021-11-09 at 1 06 38 PM" src="https://user-images.githubusercontent.com/45603961/140980263-7ea12f0f-c8a2-42dc-8b2a-be01615e2628.png">
It doesn't seem to follow the correct design patterns as the one above - unless that's intentional?

**Comment by tmitchellgcio** [comment]
These are actually two different alerts. The bottom one is from the react navigation, however I did find an article showing there is a way to keep them consistent, but that would be a completely separate ticket. Check out how production is and you will notice it's the same formatting. Nothing was in the ticket about changing the formatting.

**Comment by ajakabcin** [comment]
Oh yes I see now! ok yeah that's interesting that the formatting is different. Ok then yes this is good to go. 

I'll leave it to @andaleliz @MarciMcGuireGCIO and @Samara-Strauss to determine if addressing the navigation alert styling is something to tackle separately.

Thanks @tmitchellgcio !

**Comment by Samara-Strauss** [comment]
> These are actually two different alerts. The bottom one is from the react navigation, however I did find an article showing there is a way to keep them consistent, but that would be a completely separate ticket.

What triggers this alert? Is it a browser alert or something from the website? If it is a browser alert (eg. it shows if you go to X out of a web page), you do not have to format it to look like our on-site alerts.

**Comment by andaleliz** [comment]
That looks like a browser alert to me. I don't think we need to attempt to make any changes to that.

**Comment by tmitchellgcio** [comment]
It is a browser alert. Still I did find a way to make it like our other alerts. Nice to have but not necessary
